persistence.js

```diff
@@ -102,64 +114,37 @@ function decodeRetainedPacket (chunk, enc, cb) {
   cb(null, msgpack.decode(chunk))
 }
 
-function Sub (clientId, topic, qos) {
-  this.clientId = clientId
-  this.topic = topic
-  this.qos = qos
-}
-
 RedisPersistence.prototype.addSubscriptions = function (client, subs, cb) {
   if (!this.ready) {
     this.once('ready', this.addSubscriptions.bind(this, client, subs, cb))
     return
   }
 
-  var multi = this._db.multi()
-
-  var clientSubKey = 'client:sub:' + client.id
-  var that = this
+  var clientSubKey = clientKey + client.id
 
   var toStore = {}
-
+  var offlines = []
   var count = 0
   var published = 0
   var errored = null
 
-  multi.exists(clientSubKey)
-
   for (var i = 0; i < subs.length; i++) {
     var sub = subs[i]
     toStore[sub.topic] = sub.qos
-
     if (sub.qos > 0) {
-      var subClientKey = 'sub:client:' + sub.topic
-      var encoded = msgpack.encode(new Sub(client.id, sub.topic, sub.qos))
-      multi.rpush(subscriptionsKey, subClientKey)
-      multi.hset(subClientKey, client.id, encoded)
+      offlines.push(sub.topic)
       count++
-      that._waitFor(client, sub.topic, finish)
+      // TODO don't wait the client an extra tick
```

Comment:
Despite mosca, aedes is publishing sub topics, and even the local process is waiting for the subscribed handler (inside cached-persistence). My points:
1) to call the callback sooner, we may go mosca way and check broker.id inside message handler. Or we may even call the callback without this check and let the subscription take effect in background after client subacked !?

2) for subscription publication between nodes, cached-persistence is using broker's pub/sub. Could this be handled any lower level, not to pollute MQTT pub/sub with inter-node messaging!?